dotcom-rendering/src/lib/getTagFrontsAdPositions.ts

```diff
@@ -0,0 +1,71 @@
+import type { GroupedTrailsBase } from '../types/tagFront';
+import { getMerchHighPosition } from './getFrontsAdPositions';
+
+/**
+ * The maximum number of fronts-banner ads that can be inserted on any front.
+ */
+const MAX_FRONTS_BANNER_ADS = 6;
+
+const hasAdjacentCommercialContainer = (
+	collectionIndex: number,
+	merchHighPosition: number,
+) => collectionIndex === merchHighPosition;
+
+const isEvenIndex = (_collection: unknown, index: number) => index % 2 === 0;
+
+/**
+ * Uses a very similar approach to pressed fronts, except we
+ * - Do not need to consider thrashers
+ * - Do not need to consider the 'most viewed' container
+ *
+ * The types are also slightly different, as we no longer have
+ * specific container IDs, so we use the date which is unique
+ */
+const getTagFrontMobileAdPositions = (
+	collections: Array<GroupedTrailsBase>,
+): number[] => {
+	const merchHighPosition = getMerchHighPosition(collections.length);
+
+	return (
+		collections
+			.filter(
+				(_, index) =>
+					!hasAdjacentCommercialContainer(index, merchHighPosition),
+			)
+			.filter(isEvenIndex)
+			.map((collection) =>
+				collections.findIndex(
+					({ day, month, year }) =>
+						day === collection.day &&
+						month === collection.month &&
+						year === collection.year,
+				),
```

Comment:
This [can technically return `-1`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#return_value). Do you want to filter for it?